pkg/k8shandler/kibana/route.go

```diff
@@ -185,3 +191,50 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleExternalLogLink(
 	}
 	return nil
 }
+
+func (clusterRequest *KibanaRequest) removeSharedConfigMapPre45x() error {
+	cluster := clusterRequest.cluster
+
+	sharedConfig := NewConfigMap("sharing-config", cluster.GetNamespace(), map[string]string{})
+	err := clusterRequest.Delete(sharedConfig)
+	if err != nil && !errors.IsNotFound(err) {
+		return fmt.Errorf("Failure deleting Kibana route shared config: %v", err)
+	}
+
+	sharedRole := NewRole(
+		"sharing-config-reader",
+		cluster.Namespace,
+		NewPolicyRules(
```

Comment:
I believe you can delete by simply using the name.  This would mean there is no need to new a complete object with all the policy rules.